src/nvim/ex_getln.c

```diff
@@ -2942,10 +2942,9 @@ static int cmd_startcol(void)
 int cmd_screencol(int bytepos)
 {
   int m;  // maximum column
-
   int col = cmd_startcol();
   if (KeyTyped) {
-    m = Columns * Rows;
+    m = cmdline_win ? cmdline_win->w_width_inner * cmdline_win->w_height_inner : Columns * Rows;
```

Comment:
Would it make sense to *always* define `cmdline_win`? The default impl would anchor its dimensions to `Columns * Rows`, but that impl would be contained in the default `cmdline_win` handler, so code never needs to check if `cmdline_win` exists:

```suggestion
    m = cmdline_win->w_width_inner * cmdline_win->w_height_inner;
```

I don't know if this is a net gain, so this is suggestion is also a question :)

Reply:
Don't think so; it would have to be a grid not a window (only the default grid relates to `Columns/Rows`). But it's not straightforward to get the inner dimensions from the grid, and we can't query the cmdline row/column from the default grid, so would need ternaries anyways.